tket2-hseries/src/extension/random.rs

```diff
@@ -0,0 +1,271 @@
+                vec![int_type(6).into()],
+                EXTENSION_ID,
+                TypeBound::Any,
+                extension_ref,
+            ),
+        }
+        .into()
+    }
+}
+
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    serde::Serialize,
+    serde::Deserialize,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    EnumIter,
+    IntoStaticStr,
+    EnumString,
+    Display,
+)]
+/// The operations provided by the random extension.
+pub enum RandomOp {
+    /// `fn random_int(RNGContext) -> (RNGContext, u32)`
+    RandomInt,
+    /// `fn random_float(RNGContext) -> (RNGContext, f32)`
+    RandomFloat,
+    /// `fn random_int_bounded(RNGContext, bound: u32) -> (RNGContext, u32)`
+    RandomIntBounded,
+    /// `fn new_rng_context(seed: u64) -> Option<RNGContext>` // return None on second call
+    NewContext,
+    /// `fn delete_rng_context(RNGContext) -> ()`
+    DeleteContext,
+}
+
+impl MakeOpDef for RandomOp {
+    fn init_signature(&self, extension_ref: &std::sync::Weak<Extension>) -> SignatureFunc {
+        match self {
+            RandomOp::RandomInt => Signature::new(
+                vec![RandomType::RNGContext.get_type(extension_ref)],
+                vec![RandomType::RNGContext.get_type(extension_ref), int_type(5)],
+            ),
+            RandomOp::RandomFloat => Signature::new(
+                vec![RandomType::RNGContext.get_type(extension_ref)],
+                vec![
+                    RandomType::RNGContext.get_type(extension_ref),
+                    float64_type(),
+                ],
+            ),
+            RandomOp::RandomIntBounded => Signature::new(
+                vec![RandomType::RNGContext.get_type(extension_ref), int_type(5)],
+                vec![RandomType::RNGContext.get_type(extension_ref), int_type(5)],
+            ),
+            RandomOp::NewContext => Signature::new(
+                vec![int_type(6)],
+                Type::from(option_type(RandomType::RNGContext.get_type(extension_ref))),
+            ),
+            RandomOp::DeleteContext => {
+                Signature::new(vec![RandomType::RNGContext.get_type(extension_ref)], vec![])
+            }
+        }
+        .into()
+    }
+
+    fn from_def(op_def: &OpDef) -> Result<Self, hugr::extension::simple_op::OpLoadError> {
+        try_from_name(op_def.name(), op_def.extension_id())
+    }
+
+    fn extension(&self) -> ExtensionId {
+        EXTENSION_ID
+    }
+
+    fn extension_ref(&self) -> std::sync::Weak<Extension> {
+        Arc::downgrade(&EXTENSION)
+    }
+
+    fn description(&self) -> String {
+        match self {
+            RandomOp::RandomInt => "Generate a random integer.",
+            RandomOp::RandomFloat => "Generate a random float.",
+            RandomOp::RandomIntBounded => "Generate a random integer within the range [0, bound).",
+            RandomOp::NewContext => "Get a new RNG context with a seed.",
+            RandomOp::DeleteContext => "Discard RNG context.",
```

Comment:
Some inconsistency between names: `RandomOp::NewContext` vs `add_new_rng_context` and `RandomOp::DeleteContext` vs `add_delete_rng_context`. Do we want to add `Rng` in these names?